sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClientBuilder.java

```diff
@@ -218,6 +220,10 @@ public TableClientBuilder() {
     public TableClient buildClient() {
         TableServiceVersion serviceVersion = version != null ? version : TableServiceVersion.getLatest();
 
+        if (audience == null) {
+            audience = TableAudience.AZURE_STORAGE_PUBLIC_CLOUD;
```

Comment:
This is modifying the builder itself, which isn't what we want when creating the client.

Reply:
Also, is defaulting to Storage public cloud scope the correct choice here? If it is, please add comments explaining why this is the right choice.

Reply:
This is an unnecessary API change that I accidentally left in when figuring out how to get the audience to the BuilderHelper. @alzimmermsft Thanks for catching this, this would've been a bug. I have logic to parse the proper audience from the endpoint in the BuilderHelper but this would have circumvented that.